lib/rubocop/cop/performance/intersection_check.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # In Ruby 3.1, `Array#intersect?` has been added.
+      #
+      # This cop identifies places where `(receiver & argument).any?`
+      # can be replaced by `receiver.intersect?(argument)`.
+      #
+      # The `receiver.intersect?(argument)` method is faster than
+      # `(receiver & argument).any?`.
+      #
+      # @example
+      #   # bad
+      #   (receiver & argument).present?
+      #   (receiver & argument).any?
+      #   (receiver & argument).blank?
+      #   (receiver & argument).empty?
+      #
+      #   # good
+      #   receiver.intersect?(argument)
+      #   !receiver.intersect?(argument)
+      class IntersectionCheck < Base
```

Comment:
I'd like to avoid using the name "Check" as much as possible. What about names like "IntersectPredicate"?

```suggestion
      class IntersectPredicate < Base
```

Reply:
It's more clear, thank you.